src/gsy_e/models/market/two_sided.py

```diff
@@ -421,6 +421,9 @@ def validate_bid_offer_match(
             raise InvalidBidOfferPairException("Not all bids and offers exist in the market.")
         bid_energy = recommendation.bid_energy
         offer_energy = market_offer.energy
+        if selected_energy <= 0:
+            raise InvalidBidOfferPairException(
+                f"Energy traded {selected_energy} should be more than 0.")
```

Comment:
- I believe `selected_energy` should not be less than 0 because of the fact that [this](https://github.com/gridsingularity/gsy-e/blob/master/src/gsy_e/models/market/one_sided.py#L280-L281) part of the code is avoiding zero energies (We also get division by zero in some places).

- It should not be negative as well because it is also prohibited in the [code](https://github.com/gridsingularity/gsy-e/blob/master/src/gsy_e/models/market/one_sided.py#L112-L113).

Reply:
For the time being, we forbid negative prices in energy trading despite being a case in the real world. In order to be prepared for any coming change in the future I would maintain the if check, but I would raise a different type of exception, e.g. `NegativeTradePriceException` in order to easily identify it in the future.